File: 2021/01/31/jh20s/contest#225/1736.cpp
class Solution {
public:
string maximumTime(string time) {
	if (time[0] == '?')
		time[0] = (time[1] !='?' && time[1] >= '4') ? '1' : '2';
	if (time[1] == '?')
		time[1] =  time[0] <='1' ? '9':'3';
	if (time[3] == '?')
		time[3] = '5';
	if (time[4] == '?')
		time[4] = '9';
	return time;
}
};

File: 2021/01/31/jh20s/contest#225/1736.py
class Solution(object):
    def maximumTime(self, time):
        time = list(time)
        if time[0] =='?':
            time[0] = '1' if (time[1] !='?' and time[1] >='4') else '2'
        if time[1] =='?':
            time[1] = '9' if (time[0] <='1') else '3'
        if time[3] =='?':
            time[3] = '5'
        if time[4] =='?':
            time[4] = '9'
        return "".join(time)